language/diem-framework/modules/0L/MakeWhole.move

```diff
@@ -20,139 +22,102 @@ address 0x1 {
-                    }
-                );
+            let addr = Signer::address_of(account);
+            let cred = Credit {
+              incident_name,
+              claimed: false,
+              coins: Diem::mint<GAS>(vm, value),
             };
-        }
 
-        // add a custom list for testing purposes only. Requires vm as signer
-        public fun make_whole_test(vm: &signer, payees: vector<address>, amounts: vector<u64>){
-            CoreAddresses::assert_diem_root(vm);
-            if (!exists<Payments>(CoreAddresses::DIEM_ROOT_ADDRESS())) {
-                let i = 0;
-                let total = 0;
-                let paid = Vector::empty<bool>();
-
-                while (i < Vector::length<u64>(&amounts)) {
-                    total = total + *Vector::borrow<u64>(&amounts, i);
-                    i = i + 1;
-                    Vector::push_back<bool>(&mut paid, false);
-                };
-
-                let coins = Diem::mint<GAS>(vm, total);
-
-                move_to<Payments>(
-                    vm, 
-                    Payments{
-                        payees: payees, 
-                        amounts: amounts, 
-                        paid: paid,
-                        coins: coins
-                    }
-                );
-            };
+            if (!exists<Balance>(addr)) {
+                move_to<Balance>(account, Balance {
+                  credits: Vector::singleton(cred),
+                });
+            } else {
+              let c = borrow_global_mut<Balance>(addr);
+              Vector::push_back<Credit>(&mut c.credits, cred);
+            }
         }
 
+
         /// claims the make whole payment and returns the amount paid out
         /// ensures that the caller is the one owed the payment at index i
-        public fun claim_make_whole_payment(account: &signer, i: u64): u64 acquires Payments{
-            // find amount
+        public fun claim_make_whole_payment(account: &signer): u64 acquires Balance {
             let addr = Signer::address_of(account);
-            let payments = borrow_global_mut<Payments>(
-                CoreAddresses::DIEM_ROOT_ADDRESS()
-            );
-
-            //make sure sender is the one owed funds and that the funds have not been paid
-            //if i is invalid (<0 or >length) vector will throw error
-            assert(*Vector::borrow<address>(&payments.payees, i) == addr, Errors::internal(EWRONG_PAYEE));
-            assert(*Vector::borrow<bool>(&payments.paid, i) == false, Errors::internal(EALREADY_PAID));
+            if (!exists<Balance>(addr)) return 0;
 
-            let amount = *Vector::borrow<u64>(&payments.amounts, i);
+            let b = borrow_global_mut<Balance>(addr);
+            let amount = 0;
+            let i = 0;
+            while (i < Vector::length(&b.credits)){
+              let cred = Vector::borrow_mut(&mut b.credits, i);
+              amount = amount + Diem::value<GAS>(&cred.coins);
```

Comment:
You are summing the value in the loop, but also paying out each iteration of the loop. In the case where someone has multiple payments assigned (which I don't think should happen, but still), they will get a higher payout that they are supposed to (and the fund will be depleted before paying everyone out).

Reply:
Nice catch!

Reply:
This was spillover from a previous sketch I made. Fixed now.